lms/djangoapps/course_api/blocks/transformers/__init__.py

```diff
@@ -63,5 +64,10 @@ def __init__(
         'merged_visible_to_staff_only',
         VisibilityTransformer,
         requested_field_name='visible_to_staff_only',
+    ),
+    SupportedFieldType(
+        BlockCompletionTransformer.COMPLETION,
+        BlockCompletionTransformer,
+        'completion'
```

Comment:
*Future note to self (and others):* It would be great if we can make this pluggable, so new Transformers don't need to manually update this list - and can be automatically extracted from registered transformers.

Reply:
I remember implementing a registry for something once, and I found two basic ways of doing it: The first is by creating a registry class with a `register()` decorator function:

    @transformers.register
    class MyTransformer(BlockTransformer):

The other uses a metaclass defined on the superclass.  Then when the class gets constructed, it is automatically registered in the metaclass.  Django models use this method.

The major differences are:

1.  that the decorator is visible to the programmer, while the metaclass is hidden.
2.  programmers can opt-out of the decorator by not including it, but not the metaclass (unless the metaclass explicitly provides a hook, like writing):
        class Registerable(Registry):
            register = False
    The downside of allowing programmers to opt out (or more accurately opt not to opt-in) is that you also make it possible to *forget* to opt-in.
3.  That decorators are easier to read and write than metaclasses.